packages/openapi-v3/test/unit/controller-spec/controller-decorators/request-body-decorators/primative-types.test.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  ControllerSpec,
+  post,
+  requestBody,
+  getControllerSpec,
+} from '../../../../../';
+import {ContentObject, SchemaObject} from '@loopback/openapi-v3-types';
+import {Class} from '@loopback/repository';
+import {expect} from '@loopback/testlab';
+
+describe('requestBody decorator', () => {
+  context('for a primitive type', () => {
+    let actualSpec: ControllerSpec;
+    let expectedContent: ContentObject;
+
+    it('infers number', () => {
+      class MyController {
+        @post('/greeting')
+        greet(@requestBody() name: number) {}
+      }
+      assertRequestBodySpec({type: 'number'}, MyController);
+    });
+
+    it('infers string', () => {
+      class MyController {
+        @post('/greeting')
+        greet(@requestBody() name: string) {}
+      }
+      assertRequestBodySpec({type: 'string'}, MyController);
+    });
+
+    it('infers boolean', () => {
+      class MyController {
+        @post('/greeting')
+        greet(@requestBody() name: boolean) {}
+      }
+      assertRequestBodySpec({type: 'boolean'}, MyController);
+    });
+
+    it('infers object', () => {
+      class MyController {
+        @post('/greeting')
+        greet(@requestBody() name: object) {}
+      }
+      assertRequestBodySpec({type: 'object'}, MyController);
+    });
+
+    it('infers array', () => {
```

Comment:
This is not a valid use case ^

> Show we force the users to use @requestBody.array for the cases with array?

Yes!

Reply:
@shimks I am summarizing some features/improve we want for `@requestBody`, sorry I don't want to keep adding features in this PR since it blocks the 4th PR(upgrade the whole project), after land this one people can work on the improvement and upgrade in parallel.